File: turtle_key_client/src/node.cpp
#include "ros/ros.h"
#include "geometry_msgs/Twist.h"
#include "turtlesim/SetPen.h"

geometry_msgs::Twist output;

void callback(const geometry_msgs::Twist::ConstPtr &input) {
	output = *input;
}

int main(int argc, char **argv) {
	//Initiate ROS
	ros::init(argc, argv, "node");

	int r, g, b, width, off;

	if (argc != 6) {
		ROS_ERROR("Wrong number of arguments!!!");
		ROS_ERROR("%s [r g b width off]", argv[0]);
		return 1;
	}

	ros::NodeHandle nh;
	ros::Subscriber sub=nh.subscribe("/cmd_vel", 100, callback);
	ros::Publisher pub=nh.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel", 100);

	ros::ServiceClient clt=nh.serviceClient<turtlesim::SetPen>("/turtle1/set_pen");

	turtlesim::SetPen setpen;

	setpen.request.r = atoll(argv[1]);
	setpen.request.g = atoll(argv[2]);
	setpen.request.b = atoll(argv[3]);
	setpen.request.width = atoll(argv[4]);
	setpen.request.off = atoll(argv[5]);


	if(clt.call(setpen)) {
		ROS_INFO("Requesting %d, %d, %d, %d, %d", setpen.request.r, setpen.request.g, setpen.request.b, setpen.request.width, setpen.request.off);
		ROS_INFO("Response: %d", setpen.response);
	}
	else {
		ROS_ERROR("Failed to call service test_service");
		return 1;
	}	

	ros::Rate loop_rate(10);

	while(ros::ok()) {		
		pub.publish(output);
		loop_rate.sleep();
		ros::spinOnce();
	}
	return 0;
}

File: turtle_key_client_py/scripts/node.py
#!/usr/bin/env python
import rospy
import sys
from geometry_msgs.msg import Twist
from turtlesim.srv import SetPen

DIRECTION = Twist()

def callback(input):
    global DIRECTION
    DIRECTION = input

def usage():
    return "Wrong number of arguments!!!\n%s\n[r g b width off]"%sys.argv[0]

def key_pen(r, g, b, width, off):
    rospy.init_node('key_pen', anonymous=True)
    rospy.wait_for_service('turtle1/set_pen')  # wait for set_pen service

    request = SetPen()

    request.r = r
    request.g = g
    request.b = b
    request.width = width
    request.off = off

    try:
        response = rospy.ServiceProxy('turtle1/set_pen', SetPen)(r, g, b, width, off)
        print("Requesting r:%s, g:%s, b:%s, width:%s, off:%s"%(r, g, b, width, off))
        print("Result: %s"%response) # response not defined

    except rospy.ServiceException as e:
        print("Service call failed: %s"%e)

    finally: # run regardless status of service
        sub = rospy.Subscriber('cmd_vel', Twist, callback)
        pub = rospy.Publisher('turtle1/cmd_vel', Twist, queue_size=100)
        rate = rospy.Rate(10) # 10hz

        global DIRECTION

        while not rospy.is_shutdown():
            pub.publish(DIRECTION)
            rate.sleep()  

if __name__ == '__main__':
    try:
        if len(sys.argv) == 6:
            r = int(sys.argv[1])
            g = int(sys.argv[2])
            b = int(sys.argv[3])
            width = int(sys.argv[4])
            off = int(sys.argv[5])
            key_pen(r, g, b, width, off)
        else:
            print(usage())
            sys.exit(1)
    except rospy.ROSInterruptException:
        pass
